### PR TITLE
Resolves #215 - TYL/THX counter extended for search.php

### LINKED ISSUE #215: Extend displaying of total thx/likes in the first post for search page
#214 but for search.php page

### PR DESCRIPTION
**Pull request by Eldenroot**
Resolves #215

### CODE REVIEW

**Comment by Eldenroot** [comment]
@lairdshaw 

OK, here is the list what works fine:
- add new setting
- add new template changes
- add new lang var

But I am quite lost with this one:
image

It is because of this part of code (if I am right).
https://github.com/mybbgroup/Thank-you-like-system/pull/251/files#diff-61561fd3f118cc16cdc5c0bbb79b38c7R2046

Hmm, needs to investigate it further...

**Comment by lairdshaw** [comment]
> But I am quite lost with this one

The problem seems to be that you are hooking in to `search_end`, which is the end of the display of the search form - and since there is no thread cache for the search form, the $pids variable that you generate is empty, which the database server rightly complains about.

You need instead to be hooking in to `search_results_thread`, where the HTML for the display of each thread in the search results is generated. A word of warning though: the equivalent thread cache variable in that part of the code has an underscore - it is `$thread_cache` - so you will need to adjust your code for that.

Also, a suggestion: rather than duplicating large portions of code in the `thankyoulike_threads_udetails()` function, try to reuse code. Preferably, detect which of the two hooks called the function (e.g., if the `search_results_thread` hook called the function, then the condition `$mybb->input['action'] == "results"` will be true; if the other hook called the function, this condition  will be false), and conditionally avoid code specific to the other hook. Don't worry if this is too stressful for you though - I'd be happy to clean it up after you've finished and got it working.

**Comment by Eldenroot** [comment]
@lairdshaw - I will need more time for thinking.

There is something wrong with cache, maybe ids. So it will need more investigation from my side. I will try to do my best, this is planned for 3.3.8 so we are fine with releasing 3.3.7 which is prepared.

I will ping you if I will be out of ideas :)

Thank you buddy!

File: inc/plugins/thankyoulike.php [hunk 1]
```diff
@@ -2014,6 +2032,42 @@ function thankyoulike_threads_udetails()
...
+			if ($mybb->settings[$prefix.'thankslike'] == "like")
+			{
+				$lang->tyl_firstpost_tyl_count_search_page = $lang->sprintf($lang->tyl_firstpost_tyl_count_search_page, $lang->tyl_firstpost_tyl_count_likes);
+			}
+			elseif ($mybb->settings[$prefix.'thankslike'] == "thanks")
+			{
+				$lang->tyl_firstpost_tyl_count_search_page = $lang->sprintf($lang->tyl_firstpost_tyl_count_search_page, $lang->tyl_firstpost_tyl_count_thanks);
+			}
+
+			$tyl_forumdisplay_cached;
```
Comment (chack1172) [inline inc/plugins/thankyoulike.php:2063]: What is this?

**Comment by Eldenroot** [comment]
@lairdshaw - should be ready to review.

@chack1172 - thank you for your input!

Maybe we can extend it also for search posts, or it would be resources hungry?

**Review by lairdshaw — COMMENTED** [review]
Good work on putting this together, @Eldenroot. Mostly, it seems to be working as expected, however, it does have a bug which needs fixing, and a couple of redundant execution paths (which relate to the bug), and I also have a related suggestion:

1. The bug is present in two circumstances:
   1. When "Enable thanks/likes counter for the first post in forumdisplay_thread template?" is set to "No" and "Enable thanks/likes counter for the first post in search_results_threads_thread template?" is set to "Yes", and a forum is viewed.
   2. When "Enable thanks/likes counter for the first post in forumdisplay_thread template?" is set to "Yes", and thread-based search results are viewed (in this case, it doesn't matter what "Enable thanks/likes counter for the first post in search_results_threads_thread template?" is set to.
   In both cases, we get that familiar SQL error which you fought with earlier. See if you can figure out why and how to fix it - I'm ready to help if you need it.
2. The redundant execution paths can be seen in this light: if both settings are set to "Yes", then, when viewing forums, the code for the thread-based search results page will also be (redundantly) executed; when viewing a thread-based search result page, the code for forums will also be (redundantly) executed. (This might give you a clue about, and some insight into, the bug).
3. I suggest that duplicate code in the `thankyoulike_threads_udetails()` function be eliminated. This could be done while fixing the bug and eliminating redundant execution paths as in 1 and 2 above.

That's all for now. Good work again on coming up with this new feature!

**Comment by Eldenroot** [comment]
@lairdshaw - thank you for your review and bug report. You are right, it needs to be fixed. Anyway I am quite not sure how to do that easily. Maybe some kind of voodoo and black magic will help alongside with my basic php knowledge :) I have few ideas but none of them is the right one... (at least I think my ideas will make it even more complicated) 🗡️

**Comment by lairdshaw** [comment]
It seems to be working without error now. Ready to merge? (It would still be nice to eliminate redundant code, but if you like, you could leave that to me to tidy up after the merge.)

**Comment by Eldenroot** [comment]
It is ready to merge, please can you tidy up a little bit? Thank you buddy

**Pull request merged**